timelock-impl/src/main/java/com/palantir/atlasdb/timelock/paxos/PaxosTimestampBoundStore.java

```diff
@@ -279,14 +282,18 @@ public synchronized void storeUpperLimit(long limit) throws MultipleRunningTimes
      */
     private void checkAgreedBoundIsOurs(long limit, long newSeq, PaxosValue value) throws NotCurrentLeaderException {
         if (!proposer.getUuid().equals(value.getLeaderUUID())) {
-            String errorMsg = String.format(
-                    "Timestamp limit changed from under us for sequence '%s' (proposer with UUID '%s' changed"
-                            + " it, our UUID is '%s'). This suggests that we have lost leadership, and another timelock"
+            throwNotCurrentLeaderException(
+                    "Timestamp limit changed from under us for sequence {} (proposer with UUID '{}' changed"
+                            + " it, our UUID is '{}'). This suggests that we have lost leadership, and another timelock"
                             + " server has gained leadership and updated the timestamp bound."
-                            + " The offending bound was '%s'; we tried to propose"
-                            + " a bound of '%s'. (The offending Paxos value was '%s'.)",
-                    newSeq, value.getLeaderUUID(), proposer.getUuid(), PtBytes.toLong(value.getData()), limit, value);
-            throwNotCurrentLeaderException(errorMsg);
+                            + " The offending bound was '{}'; we tried to propose"
+                            + " a bound of '{}'. (The offending Paxos value was '{}'.)",
+                    SafeArg.of("sequence", newSeq),
```

Comment:
I'm not sure on the naming of these args, so as always, will take suggestions

Reply:
these are good!